Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -623,11 +637,154 @@ - (void)fetchProfileForRecipientId:(NSString *)recipientId
     }
 
     userProfile.lastUpdateDate = [NSDate new];
-    [self.dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
-        [userProfile saveWithTransaction:transaction];
-    }];
 
-    // TODO: Actually update the profile.
+    [self saveUserProfile:userProfile];
+
+    [ProfileFetcherJob runWithRecipientId:recipientId
+                           networkManager:self.networkManager
+                         ignoreThrottling:ignoreThrottling];
+}
+
+- (void)updateProfileForRecipientId:(NSString *)recipientId
+               profileNameEncrypted:(NSData *_Nullable)profileNameEncrypted
+                 avatarUrlEncrypted:(NSData *_Nullable)avatarUrlEncrypted
```

Comment:
As discussed I think the URL/digest will be plain text, but the blob will be encrypted. Though we can address that as part of the decryption PR.